modules/contact-form/js/gutenblock.jsx

```diff
@@ -0,0 +1,630 @@
+/*global wp*/
+/** @jsx wp.element.createElement */
+/** @format */
+
+/**
+ * External dependencies
+ */
+import {
+	PanelBody,
+	TextControl,
+	TextareaControl,
+	CheckboxControl,
+	ToggleControl,
+	IconButton
+} from '@wordpress/components';
+
+import {
+	registerBlockType,
+	createBlock
+} from '@wordpress/blocks';
+
+import {
+	InnerBlocks,
+	InspectorControls
+} from '@wordpress/editor';
+
+import {
+	Component,
+	Fragment
+} from '@wordpress/element';
+
+import {
+	__
+} from '@wordpress/i18n';
+
+/**
+ * Components:
+ */
+
+class GrunionForm extends Component {
+	constructor( ...args ) {
+		super( ...args );
+		this.onChangeSubject = this.onChangeSubject.bind( this );
+		this.onChangeTo = this.onChangeTo.bind( this );
+		this.onChangeSubmit = this.onChangeSubmit.bind( this );
+	}
+
+	onChangeSubject( x ) {
+		this.props.setAttributes( { subject: x } );
+	}
+
+	onChangeTo( x ) {
+		this.props.setAttributes( { to: x } );
+	}
+
+	onChangeSubmit( x ) {
+		this.props.setAttributes( { submit_button_text: x } );
+	}
+
+	render() {
+		return (
+			<Fragment>
+				<InspectorControls>
+					<PanelBody title={ __( 'Submission Details', 'jetpack' ) }>
+						<TextControl
+							label={ __( 'What would you like the subject of the email to be?' ) }
+							value={ this.props.subject }
+							onChange={ this.onChangeSubject }
+						/>
+						<TextControl
+							label={ __( 'Which email address should we send the submissions to?' ) }
+							value={ this.props.to }
+							onChange={ this.onChangeTo }
+						/>
+						<TextControl
+							label={ __( 'What should the label on the form’s submit button say?' ) }
+							value={ this.props.submit_button_text }
+							placeholder={ __( 'Submit »' ) }
+							onChange={ this.onChangeSubmit }
+						/>
+					</PanelBody>
+				</InspectorControls>
+				<div className="grunion-form">
+					{this.props.children}
+					<TextControl
+						className="button button-primary button-default grunion-submit-button"
+						value={ this.props.submit_button_text }
```

Comment:
Nit; this should be camelCased

Reply:
This is intentionally not camelCased, as it's an attribute drawn from the legacy shortcode and it simplifies migrations by using the same attribute names for the shortcode and block.

Reply:
Ah! Interesting. What would be a reason not to rename the attribute name in the shortcode transform function? I.e. how does it simplify the process?

Reply:
By leaving the variable name the same, we both make the transformation process from shortcode to block simpler, as well as retain the ability to use a common interpreter for the front-end to render the form, whether it's a shortcode or a block.